app/pages/project/classify.jsx

```diff
@@ -0,0 +1,434 @@
+      subject = Promise.resolve(subjectToLoad);
+    }
+
+    // If there aren't any left (or there weren't any to begin with), refill the list.
+    if (upcomingSubjects.forWorkflow[workflow.id].length === 0) {
+      // console.log 'Fetching subjects', workflow.id
+      this.maybePromptWorkflowAssignmentDialog(this.props);
+
+      const subjectQuery = { workflow_id: workflow.id };
+
+      if (subjectSet) {
+        subjectQuery.subject_set_id = subjectSet.id;
+      }
+
+      const fetchSubjects = apiClient.get('/subjects/queued', subjectQuery).catch((error) => {
+        if (error.message.indexOf('please try again') === -1) {
+          throw error;
+        } else {
+          return new Promise((resolve, reject) => {
+            const fetchSubjectsAgain = (() => apiClient.get('/subjects/queued', subjectQuery).then(resolve).catch(reject));
+            setTimeout(fetchSubjectsAgain, 2000);
+          });
+        }
+      }).then((subjects) => {
+        const nonLoadedSubjects = subjects.filter(newSubject => newSubject !== subjectToLoad);
+        const filteredSubjects = nonLoadedSubjects.filter(nonLoadedSubject => !nonLoadedSubject.already_seen && !nonLoadedSubject.retired);
+        const subjectsToLoad = (filteredSubjects.length > 0) ? filteredSubjects : nonLoadedSubjects;
+
+        upcomingSubjects.forWorkflow[workflow.id].push(...subjectsToLoad);
+
+        // Remove any duplicate subjects from the upcoming queue
+        return upcomingSubjects.forWorkflow[workflow.id].filter((upcomingSubject, idx) =>
+          upcomingSubjects.forWorkflow[workflow.id].indexOf(upcomingSubject) === idx
+        );
+      });
+
+      // If we're filling this list for the first time, we won't have a subject selected, so try again.
+      if (!subject) {
+        subject = fetchSubjects.then(() => {
+          if (upcomingSubjects.forWorkflow[workflow.id].length === 0) {
+            // TODO: If this fails during a random workflow, pick the next workflow.
+            throw new Error(`No subjects available for workflow ${workflow.id}`);
+          } else {
+            return upcomingSubjects.forWorkflow[workflow.id].shift();
+          }
+        });
+      }
+
+      // TODO: Pre-load images for the next subject.
+    }
+
+    // console.log 'Chose a subject'
+    return subject;
+  },
+
+  render() {
+    return (
+      <div className="classify-page content-container">
+        <Helmet title={`${this.props.project.display_name} » ${counterpart('classifyPage.title')}`} />
+
+        {this.props.projectIsComplete &&
+          <FinishedBanner project={this.props.project} />}
+
+        {this.state.validUserGroup &&
+          <p className="anouncement-banner--group">You are classifying as a student of your classroom.</p>}
+
+        {this.renderClassifier()}
+      </div>
+    );
+  },
+
+  renderClassifier() {
+    if (this.state.classification) {
+      return (
+        <Classifier.default
```

Comment:
Why is this `Classifier.default` and not `Classifier`?

Reply:
That's how I had it originally, but it didn't work (React fails to render, and says that the page is trying to render something that's not a React component). I haven't a clue why not.

Reply:
The Classifier being imported, index.cjsx in the classifier folder, is still coffeescript, so it's using the CommonJS modules rather than the newer spec. You can switch the import to a CommonJS require instead and specify default there:
`const Classifier = require('../../classifier').default;`

Reply:
My bad. I broke that export when I connected the classifier to the Redux store.

Reply:
Should be fixed in #4429.

Reply:
#4429 is now merged.